atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepMetrics.java

```diff
@@ -0,0 +1,93 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.sweep;
+
+import org.mpierce.metrics.reservoir.hdrhistogram.HdrHistogramReservoir;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricRegistry;
+import com.palantir.atlasdb.keyvalue.api.SweepResults;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.util.AtlasDbMetrics;
+
+class SweepMetrics {
+    private static final String STALE_VALUES_DELETED = "staleValuesDeleted";
+    private static final String CELLS_EXAMINED = "cellsExamined";
+    private static final String AGGREGATE_STALE_VALUES_DELETED = MetricRegistry.name(SweepMetrics.class,
+            STALE_VALUES_DELETED);
+    private static final String AGGREGATE_CELLS_EXAMINED = MetricRegistry.name(SweepMetrics.class,
+            CELLS_EXAMINED);
+
+    private final MetricRegistry metricRegistry;
+
+    public static SweepMetrics create() {
+        SweepMetrics sweepMetrics = new SweepMetrics(AtlasDbMetrics.getMetricRegistry());
+        sweepMetrics.registerAggregateMetrics();
+        return sweepMetrics;
+    }
+
+    protected SweepMetrics(MetricRegistry metricRegistry) {
+        this.metricRegistry = metricRegistry;
+    }
+
+    private void registerAggregateMetrics() {
+        registerMetricWithHdrHistogram(AGGREGATE_STALE_VALUES_DELETED);
+        registerMetricWithHdrHistogram(AGGREGATE_CELLS_EXAMINED);
+    }
+
+    void registerMetricsIfNecessary(TableReference tableRef) {
+        registerMetricWithHdrHistogram(getCellsDeletedMetric(tableRef));
+        registerMetricWithHdrHistogram(getCellsExaminedMetric(tableRef));
+    }
+
+    void recordMetrics(TableReference tableRef, SweepResults results) {
+        recordCellsDeleted(tableRef, results.getCellsDeleted());
+        recordCellsExamined(tableRef, results.getCellsExamined());
+    }
+
+    private void recordCellsDeleted(TableReference tableRef, long cellsDeleted) {
+        String deletesMetric = getCellsDeletedMetric(tableRef);
+
+        metricRegistry.histogram(deletesMetric).update(cellsDeleted);
+        metricRegistry.histogram(AGGREGATE_STALE_VALUES_DELETED).update(cellsDeleted);
+    }
+
+    private void recordCellsExamined(TableReference tableRef, long cellsExamined) {
+        String examinedMetric = getCellsExaminedMetric(tableRef);
+
+        metricRegistry.histogram(examinedMetric).update(cellsExamined);
+        metricRegistry.histogram(AGGREGATE_CELLS_EXAMINED).update(cellsExamined);
+    }
+
+    private String getCellsExaminedMetric(TableReference tableRef) {
+        return MetricRegistry.name(SweepMetrics.class, CELLS_EXAMINED, tableRef.getQualifiedName());
+    }
+
+    private String getCellsDeletedMetric(TableReference tableRef) {
```

Comment:
Almost all of the code above is duplicated. Would it make more sense to have an enum for metric types that implements all of the functions above? Seems like it would also make more sense once we have more metrics.